test/jdk/java/nio/channels/AsyncCloseAndInterrupt.java

```diff
@@ -133,11 +137,15 @@ private static void initFile() throws Exception {
             if (!fifoFile.delete())
                 throw new IOException("Cannot delete existing fifo " + fifoFile);
         }
-        Process p = Runtime.getRuntime().exec("mkfifo " + fifoFile);
-        if (p.waitFor() != 0)
-            throw new IOException("Error creating fifo");
-        new RandomAccessFile(fifoFile, "rw").close();
 
+        ProcessBuilder pb = new ProcessBuilder("mkfifo", fifoFile.toString());
+        OutputAnalyzer oa = ProcessTools.executeProcess(pb);
+        oa.waitFor();
+        if (oa.pid() != 0)
```

Comment:
The test is currently failing for me on WSL1 (Linux configuration), both with and without your changes; presumably NTFS doesn't support FIFO files. Should we fail or skip the test in this case? I'm fine with either, WSL is not a supported build config, and this is not the only failure.

If you choose to fail the test, then you can use:
```
ProcessTools.executeProcess(pb).shouldHaveExitValue(0);
```
this will dump the output and throw a RuntimeException for you.

If you choose to skip the fifo test instead, `oa.getExitValue()` is what you need here.

Reply:
The `FileStore` type is "NTFS"?

Reply:
> `oa.getExitValue()` is what you need here.

Using `oa.pid()` was lame.

Reply:
`FileStore` type is `drvfs`, which probably stands for "delegated to Windows".

Reply:
I think that means you were in a WSL (Linux) partition. On my system in jshell I think I got that when in `/home/bpb` but got "NTFS" when in `C:\Users\bpb\`.

Reply:
In 8ad1bdc138590ac52618541175a64d8a477cfc4f I modified the test to to print a warning instead of throwing an exception if `mkfifo` fails due to not being supported on the target file system. This allows most of the test to be executed.